tempodb/compactor.go

```diff
@@ -171,17 +176,18 @@ func (rw *readerWriter) compact(blockMetas []*backend.BlockMeta, tenantID string
 		iters = append(iters, iter)
 	}
 
-	recordsPerBlock := (totalRecords / outputBlocks)
+	recordsPerBlock := totalRecords / outputBlocks
 	var newCompactedBlocks []*backend.BlockMeta
 	var currentBlock *encoding.StreamingBlock
 	var tracker backend.AppendTracker
 
-	combiner := instrumentedObjectCombiner{
-		inner:                rw.compactorSharder,
-		compactionLevelLabel: compactionLevelLabel,
-	}
-
-	iter := encoding.NewMultiblockIterator(ctx, iters, rw.compactorCfg.IteratorBufferSize, combiner, dataEncoding)
+	iter := encoding.NewMultiblockIterator(ctx, iters, rw.compactorCfg.IteratorBufferSize, dataEncoding, compactionLevelLabel,
+		func(id []byte, initial []byte) model.ObjectMerger {
+			return &instrumentedObjectMerger{
+				ObjectMerger:         model.NewMerger(id, initial),
+				compactionLevelLabel: compactionLevelLabel,
```

Comment:
Passing the `compactionLevelLabel` to both the `instrumentedObjectMerger` and `NewMultiblockIterator` just for the metric is a little 😞 . I don't have a cleaner alternative, though.